stylix/autoload.nix

```diff
@@ -9,6 +9,11 @@ builtins.concatLists (
     let
       file = "${inputs.self}/modules/${path}/${for}.nix";
     in
-    lib.optional (kind == "directory" && builtins.pathExists file) file
+    lib.optional (kind == "directory" && builtins.pathExists file) (
+      if lib.functionArgs (import file) == { } then
+        lib.modules.importApply file (import ./mk-target.nix)
+      else
+        file
+    )
```

Comment:
Nit: could do with some named variables to a) increase readability and b) improve nix eval perf

Bug: you need to check `isFunction` before using `functionArgs`; a plain module would cause `error: 'functionArgs' requires a function`.

See other comment for ideas.